pkg/identityserver/organization_registry.go

```diff
@@ -243,6 +250,33 @@ func (is *IdentityServer) deleteOrganization(ctx context.Context, ids *ttnpb.Org
 	return ttnpb.Empty, nil
 }
 
+func (is *IdentityServer) purgeOrganization(ctx context.Context, ids *ttnpb.OrganizationIdentifiers) (*types.Empty, error) {
+	if !is.IsAdmin(ctx) {
+		return nil, errAdminsPurgeOrganizations
+	}
+	err := is.withDatabase(ctx, func(db *gorm.DB) error {
+		err := store.GetContactInfoStore(db).DeleteEntityContactInfo(ctx, ids)
+		if err != nil {
+			return err
+		}
+		// delete related API keys before purging the organization
+		err = store.GetAPIKeyStore(db).DeleteEntityAPIKeys(ctx, ids)
+		if err != nil {
+			return err
+		}
+		err = store.GetMembershipStore(db).DeleteAccountMembers(ctx, ids.GetOrganizationOrUserIdentifiers())
```

Comment:
```suggestion
        // Delete related memberships before purging the organization
        err = store.GetMembershipStore(db).DeleteAccountMembers(ctx, ids.GetOrganizationOrUserIdentifiers())
```

Nit: keep your commenting consistent. Also, from our [DEVELOPMENT.md](https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.10/DEVELOPMENT.md#comments) guidelines:

> All comments should be English sentences, starting with a capital letter and ending with a period.